src/link_mutator.rs

```diff
@@ -36,7 +36,17 @@ pub fn type_declarations_from_source(code: &str) -> Result<Vec<ExportedTypeDecla
 fn should_keep_default_type(type_info: &TypeInfo, resolved_types: &[String]) -> bool {
     // TODO: we could be more clever here, but for now we keep it simple
     match type_info {
-        TypeInfo::Basic(name) => resolved_types.contains(&name.token().to_string()),
+        TypeInfo::Basic(name) => {
+            let name_string = &name.token().to_string();
+            if resolved_types.contains(name_string) {
+                true
+            } else {
+                matches!(
+                    name_string.as_str(),
+                    "boolean" | "buffer" | "number" | "string" | "thread" | "unknown" | "any"
+                )
+            }
```

Comment:
Maybe resolved_types should actually contain the builtin types, and we don't need to change this part?

```suggestion
            resolved_types.contains(name_string) || matches!(
                name_string.as_str(),
                "boolean" | "buffer" | "number" | "string" | "thread" | "unknown" | "any"
            )
```